lib/ansible/modules/storage/vexata/vexata_volume.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Sandeep Kasargod (sandeep@vexata.com)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: vexata_volume
+version_added: 2.8
+short_description: Manage volumes on Vexata VX100 storage arrays
+description:
+    - Create, deletes or extend volumes on a Vexata VX100 array.
+author:
+- Sandeep Kasargod (@vexata)
+options:
+  name:
+    description:
+      - Volume name.
+    required: true
+  state:
+    description:
+      - Creates/Modifies volume when present or removes when absent.
+    default: present
+    choices: [ present, absent ]
+  size:
+    description:
+      - Volume size in M, G, T units. M=2^20, G=2^30, T=2^40 bytes.
+extends_documentation_fragment:
+    - vexata.vx100
+'''
+
+EXAMPLES = '''
+- name: Create new 2 TiB volume named foo
+  vexata_volume:
+    name: foo
+    size: 2T
+    state: present
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+
+- name: Expand volume named foo to 4 TiB
+  vexata_volume:
+    name: foo
+    size: 4T
+    state: present
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+
+- name: Delete volume named foo
+  vexata_volume:
+    name: foo
+    state: absent
+    array: vx100_ultra.test.com
+    user: admin
+    password: secret
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vexata import (
+    argument_spec, get_array, required_together, size_to_MiB)
+
+
+def get_volume(module, array):
+    """Retrieve a named volume if it exists, None if absent."""
+    name = module.params['name']
+    try:
+        vols = array.list_volumes()
+        vol = filter(lambda v: v['name'] == name, vols)
+        if len(vol) == 1:
+            return vol[0]
+        else:
+            return None
+    except Exception:
+        module.fail_json(msg='Error while attempting to retrieve volumes.')
+
+
+def create_volume(module, array):
+    """"Create a new volume."""
+    changed = False
```

Comment:
```suggestion
    changed = False
    if not module.params.get('size', False):
        module.fail_json(msg='Size is required to update volume')
```

Reply:
Refactored. Error message in review comment should be for create, not update.